### DISCUSSION TITLE
Question about quantizing term weights

### DISCUSSION THREAD

**Discussion by kwang2049** [castorini/pyserini#982] [Q&A] [answered]
Dear Pyserini/Anserini contributors,

I have some questions about how Py/Anserini deals with term weights, especially considering quantization. 

I found without quantizing the term weights of the collection (which is the input for indexing), performance will drop significantly (I tried uniCOIL and SPLADEv2). So:

1. Why does this happen (performance drops if one does not do quantization)? 
2. Does Py/Anserini only accept integer weights and would quantize the weights anyhow if it was not done by the users?

Thanks in advance for your reply:)

**Comment by MXueguang** [comment, marked as answer]
Hi @kwang2049
take uniCOIL as example, 
the term weights generated by the model are usually in range 0-5 as float.
Py/Anserini only accepts integer weights for corpus, thus we quantitize the float in range 0-5 to integers in range 0-255.
If indexing the term weights without quantization, the floating points will be rounded into integers directly.
Integer 0-5 loss many infomation while describing a document v.s. integer 0-255

**Reply by kwang2049** [reply]
Thanks so much for your quick reply!

Now I understand the doc term weights in Py/Anserini.

And then what about the query term weights? Does Py/Anserini use **float** **query** term weights and **integer** **doc** term weights and compute a **float score** in the end?

It seems that one does not need to do quantization over query terms, according to the [search method](https://github.com/castorini/pyserini/blob/02ad588aa578ab0532e32fe4e20fda24c604dc92/pyserini/search/_impact_searcher.py#L117) in the `ImpactSearcher` class.

**Reply by MXueguang** [reply]
yes, since Lucene/Anserini support float query weights, you can keep it as float while using ImpactSearcher.
